### PR TITLE
Add Task Resources: Git Clone With Acess Token

### PR DESCRIPTION
**Pull request by Andrew-X-Ying**
Add Task Resources: Git Clone With Acess Token.

### CODE REVIEW

**Comment by Andrew-X-Ying** [comment]
Hello @DavidObando @daytonellwanger , could you please help review this PR? Thanks.
Seems that I cannot add reviewer because I have no write access (https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review).
So I have to at you here.

File: Tasks/git-clone-accesstoken/main.ps1 [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    [string]$At,
+    [Parameter()]
+    [string]$Directory,
+    [Parameter()]
+    [string]$Branch
+)
+
+# ====== Create repository folder is it doesnot exist.
+
+try {
```
Comment (Qingchaogong) [inline Tasks/git-clone-accesstoken/main.ps1:14]: It is worth validating the required parameters before proceeding.

File: Tasks/git-clone-accesstoken/main.ps1 [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    git clone -b $Branch $Link
+}
+else 
+{
+    git clone -b $Link
+}
+
+Pop-Location
+
+Write-Output "Has completed cloing repostiry."
```
Comment (Qingchaogong) [inline Tasks/git-clone-accesstoken/main.ps1:80]: nit: typos in cloing and repostiry. And you may use: The repository has been successfully cloned.

File: Tasks/git-clone-accesstoken/task.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  branch:
+    default: ''
+    type: string
+    required: false
+    description: The branch to clone.
+documentation:
+  notes: |
+    This task allows you to clone a repository with access token.
+  examples:
+    - name: git-clone
```
Comment (Qingchaogong) [inline Tasks/git-clone-accesstoken/task.yaml:36]: please change the name to git-clone-accesstoken

File: Tasks/git-clone-accesstoken/task.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      description: Clone this repository into C:\Workspaces
+      parameters:
+        repositoryUrl: https://github.com/microsoft/calculator
+        at: theaccesstoken
+        directory: C:\Workspaces
+    - name: git-clone
```
Comment (Qingchaogong) [inline Tasks/git-clone-accesstoken/task.yaml:42]: Change the name to right task name.

File: Tasks/git-clone-accesstoken/main.ps1 (hunk 1 above)
Comment (Qingchaogong) [reply Tasks/git-clone-accesstoken/main.ps1:14]: Looks like our task validation in service can make sure of required parameters. So it is optional to add the validators here.

Comment (Andrew-X-Ying) [reply Tasks/git-clone-accesstoken/main.ps1:14]: Thanks for the suggestion. I think we can rely on backend check.

File: Tasks/git-clone-accesstoken/task.yaml (hunk 3 above)
Comment (Andrew-X-Ying) [reply Tasks/git-clone-accesstoken/task.yaml]: Thanks for catching this. It's fixed.

File: Tasks/git-clone-accesstoken/task.yaml (hunk 4 above)
Comment (Andrew-X-Ying) [reply Tasks/git-clone-accesstoken/task.yaml]: Thanks for catching this. It's fixed.

**Review by Qingchaogong — DISMISSED** [review]

File: Tasks/git-clone-accesstoken/main.ps1 (hunk 2 above)
Comment (Andrew-X-Ying) [reply Tasks/git-clone-accesstoken/main.ps1]: Thanks. It's fixed.

**Comment by Qingchaogong** [comment]
> Hello @DavidObando @daytonellwanger , could you please help review this PR? Thanks. Seems that I cannot add reviewer because I have no write access (https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review). So I have to at you here.

I assignee the PRs to David and Dayton :)

**Pull request closed without merging**